chain/chain/src/flat_storage_resharder.rs

```diff
@@ -200,15 +204,21 @@ impl FlatStorageResharder {
         Ok(())
     }
 
-    /// Returns an error if a resharding event is in progress.
-    fn check_no_resharding_in_progress(&self) -> Result<(), StorageError> {
-        // Do not allow multiple resharding events in parallel.
-        if self.resharding_event().is_some() {
+    /// Returns `Ok` if:
+    /// - no resharding event exists.
+    /// - a resharding event already exists, but it's not in progress yet.
+    ///
+    /// Returns `Err` if:
+    /// - a resharding event is in progress.
```

Comment:
Can you add under what circumstances this might happen and / or why do you think it should never happen?